include/zmq.h

```diff
@@ -763,6 +766,10 @@ ZMQ_EXPORT int zmq_socket_monitor_versioned (
   void *s_, const char *addr_, uint64_t events_, int event_version_, int type_);
 ZMQ_EXPORT int zmq_socket_monitor_pipes_stats (void *s);
 
+#ifdef ZMQ_HAVE_PPOLL
```

Comment:
Don't ifdef in the public header, otherwise the ABI is different depending on the build. Instead return an error OPNOTSUPP or something like that from the implementation